kernel/packages/shared/apis/RestrictedActionModule.ts

```diff
@@ -0,0 +1,63 @@
+import { exposeMethod, registerAPI } from 'decentraland-rpc/lib/host'
+import { ExposableAPI } from './ExposableAPI'
+import defaultLogger from '../logger'
+import { unityInterface } from '../../unity-interface/UnityInterface'
+import { ParcelIdentity } from './ParcelIdentity'
+import { Vector3 } from '../../decentraland-ecs/src/decentraland/math'
+import { gridToWorld, isInParcel, parseParcelPosition } from '../../atomicHelpers/parcelScenePositions'
+
+enum Permission {
+  ALLOW_MOVE_INSIDE_SCENE = 'ALLOW_MOVE_INSIDE_SCENE'
+}
+
+export interface IRestrictedActionModule {
+  requestMoveTo(newPosition: Vector3): Promise<void>
+}
+
+@registerAPI('RestrictedActionModule')
+export class RestrictedActionModule extends ExposableAPI implements IRestrictedActionModule {
+  parcelIdentity = this.options.getAPIInstance(ParcelIdentity)
+
+  getSceneData() {
+    return this.parcelIdentity.land.sceneJsonData
+  }
+
+  hasPermission(permission: Permission) {
+    const json = this.getSceneData()
+    const list = json.requiredPermissions || []
+    return list.indexOf(permission) !== -1
+  }
+
+  calculatePosition(newPosition: Vector3) {
+    const base = parseParcelPosition(this.getSceneData().scene.base)
+
+    const basePosition = new Vector3()
```

Comment:
Could this be changed as
```
    const baseWorld = gridToWorld(base.x, base.y)
    return baseWorld.add(newPosition)
```
?

Reply:
@nchamo no, we don't. If we change it we loss the method `add` because `gridToWorld` returns a `Vector3Component` and it doesn't have the method `add`.

Reply:
Oh, ok. Closing this then